staging-service/src/main/java/com/github/cafdataprocessing/services/staging/utils/JsonMinifier.java

```diff
@@ -34,6 +34,7 @@ public static final void minifyJson(final InputStream inputStream, final OutputS
         while (parser.nextToken() != null) {
             gen.copyCurrentEvent(parser);
         }
+        gen.flush();
```

Comment:
JsonGenerator is closeable and the concrete implementation WriterBasedJsonGenerator has this impl

```
    @Override
    public void close() throws IOException
    {
        super.close();

        /* 05-Dec-2008, tatu: To add [JACKSON-27], need to close open
         *   scopes.
         */
        // First: let's see that we still have buffers...
        if (_outputBuffer != null
            && isEnabled(Feature.AUTO_CLOSE_JSON_CONTENT)) {
```
While .flush may solve the problem, it's robbing us of any buffering benefits. 
Instead of writing the line sep directly to the outputstream
`outstream.write(System.lineSeparator().getBytes());`
I think you should be using gen.writeRaw(System.lineSeparator()

Reply:
Appending newline using generator

Reply:
Updated to use generator features to control stream